### PR TITLE
[FLINK-6328] [chkPts] Don't add savepoints to CompletedCheckpointStore

### PR DESCRIPTION
**Pull request by tillrohrmann**
The lifecycle of savepoints is not managed by the CheckpointCoordinator and fully
in the hand of the user. Therefore, the CheckpointCoordinator cannot rely on them
when trying to recover from failures. E.g. a user moving a savepoint shortly before
a failure could completely break Flink's recovery mechanism because Flink cannot
skip failed checkpoints when recovering.

Therefore, until Flink is able to skip failed checkpoints when recovering, we should
not add savepoints to the CompletedCheckpointStore which is used to retrieve checkpoint
for recovery. The distinction of a savepoint is done on the basis of the
CheckpointProperties (CheckpointProperties.STANDARD_SAVEPOINT).

cc @rmetzger

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java [hunk 1]
```diff
@@ -864,22 +864,28 @@ private void completePendingCheckpoint(PendingCheckpoint pendingCheckpoint) thro
...
-				// we failed to store the completed checkpoint. Let's clean up
-				executor.execute(new Runnable() {
-					@Override
-					public void run() {
-						try {
-							completedCheckpoint.discardOnFailedStoring();
-						} catch (Throwable t) {
-							LOG.warn("Could not properly discard completed checkpoint {}.", completedCheckpoint.getCheckpointID(), t);
+			// TODO: add savepoints to completed checkpoint store once FLINK-4815 has been completed
+			if (!CheckpointProperties.isSavepoint(completedCheckpoint.getProperties())) {
```
Comment (tony810430) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java]: Why not use `completedCheckpoint.getProperties().isSavepoint()` to check whether it is a savepoint or not?
This method had already been implemented in `CheckpoinProperties`.

Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java]: Good point. I missed that. Will adapt the PR.

**Comment by rmetzger** [comment]
+1 to merge

**Comment by tillrohrmann** [comment]
Thanks for the review @tony810430 and @rmetzger. Merging this PR.

**Pull request closed without merging**